docs/content/preview/integrations/hashicorp-vault.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Hashicorp Vault
+linkTitle: Hashicorp Vault
+description: Use YSQL plugin for Hashicorp Vault
+aliases:
+menu:
+  preview_integrations:
+    identifier: hashicorp-vault
+    parent: integrations
+    weight: 571
+type: docs
+---
+
+[HashiCorp Vault](https://www.hashicorp.com/products/vault) is designed to help organizations manage access to secrets and transmit them safely.
+
+Secrets are any form of sensitive credentials that need to be tightly controlled and monitored and can be used to unlock sensitive information. Secrets can be in the form of passwords, API keys, SSH keys, RSA tokens, or OTP.
+
+For more details, refer to [Dynamic secrets](https://github.com/yugabyte/hashicorp-vault-ysql-plugin#dynamic-secrets).
+
+## YSQL plugin for Hashicorp Vault
+
+YSQL plugin for Hashicorp Vault provides APIs for using HashiCorp Vault dynamic secrets with YugabyteDB. The plugin includes APIs for adding YugabyteDB to manage secrets, creating new users, and managing [leases](https://developer.hashicorp.com/vault/docs/concepts/lease).
+
+For more details, refer to [YSQL plugin for Hashicorp Vault](https://github.com/yugabyte/hashicorp-vault-ysql-plugin#ysql-plugin-for-hashicorp-vault-1).
+
+## Setup
+
+Before using the vault, do the following:
+
+- Add go to the path as follows:
+
+    ```sh
+    export GOPATH=$HOME/go
+    export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+    ```
+
+- To run the vault in development mode, add the default vault address and vault token as follows:
+
+    ```sh
+    export VAULT_ADDR="http://localhost:8200"
+    export VAULT_TOKEN="root"
+    ```
+
+Clone the `hashicorp-vault-ysql-plugin` repository:
+
+```sh
+git clone https://github.com/yugabyte/hashicorp-vault-ysql-plugin && cd hashicorp-vault-ysql-plugin
+```
+
+Build the plugin as follows:
+
+```go
+go build -o <build dir>/ysql-plugin cmd/ysql-plugin/main.go
+```
+
+Your YugabyteDB cluster should be up and running. Refer to [YugabyteDB Prerequisites](../../tools/#yugabytedb-prerequisites).
+
+## Run and configure the vault server
+
+The vault automatically registers the plugin if the directory of the binary of the plugin is provided as an input with the `dev-plugin-dir` flag. To run the vault server in development mode, use the `dev` flag. Set the `dev-root-token` flag which informs the vault to use the default vault token of root to login (this token is required in production mode).
```

Comment:
Can we rephrase this to something like : 

Vault server can be run in development mode using `dev` flag, this makes it easy to experiment with Vault or start a Vault instance for development. Vault automatically registers the plugin if the directory of the binary of the plugin is provided as an input with the `dev-plugin-dir` flag. Set the `dev-root-token` flag which informs the vault to use the default vault token of root to login (this token is required in production mode).

Also, add the following warning as mentioned in [vault docs](https://developer.hashicorp.com/vault/docs/concepts/dev-server) : 

Warning: Never, ever, ever run a "dev" mode server in production. It is insecure and will lose data on every restart (since it stores data in-memory). It is only made for development or experimentation.